src/tests/ftest/util/agent_utils.py

```diff
@@ -33,91 +33,64 @@
-    if node_server_type == NodeListType.SERVER:
-        socket_dir = "/var/run/daos_server"
-    elif node_server_type == NodeListType.CLIENT:
-        socket_dir = "/var/run/daos_agent"
-    else:
-        raise AgentFailed("Unknown node type, exiting")
-
-    okay = True
-    failed_node = None
-    for node in node_list:
-        cmd = "test -d " + socket_dir
-        resp = subprocess.call(["ssh", node, cmd])
-        if resp != 0:
-            okay = False
-            failed_node = node
-            break
-    return okay, failed_node, socket_dir
-
-# Disabling check for this function as in this case, more variables is more
-# clear than encapsulating in dict or object
 def run_agent(basepath, server_list, client_list=None):
-    """
-    Makes sure the environment is setup for the security agent and then launches
+    """Start daos agents on the specified hosts.
+
+    Make sure the environment is setup for the security agent and then launches
     it on the compute nodes.
 
     This is temporary; presuming the agent will deamonize at somepoint and
     can be started killed more appropriately.
 
-    basepath    --root directory for DAOS repo or installation
-    client_list --those nodes that are acting as compute nodes in the test
-    server_list --those nodes that are acting as server nodes in the test
+    Args:
+        basepath (str): root directory for DAOS repo or installation
+        server_list (list): nodes acting as server nodes in the test
+        client_list (list, optional): nodes acting as client nodes in the test.
+            Defaults to None.
+
+    Raises:
+        AgentFailed: if there is an error starting the daos agents
+
+    Returns:
+        dict: set of subprocess sessions
+
     """
     sessions = {}
-
     user = getpass.getuser()
 
-    retcode, node, agent_dir = node_setup_okay(server_list, NodeListType.SERVER)
-    if not retcode:
-        raise AgentFailed("Server node " + node + " does not have directory "
-                          + agent_dir + " set up correctly for user "
-                          + user + ".")
-
     # if empty client list, 'self' is effectively client
     if client_list is None:
         client_list = [socket.gethostname().split('.', 1)[0]]
 
-    retcode, node, agent_dir = node_setup_okay(client_list, NodeListType.CLIENT)
-    if not retcode:
-        raise AgentFailed("Client node " + node + " does not have directory "
-                          + agent_dir + " set up correctly for user "
-                          + user + ".")
+    file_checks = (
+        ("Server", server_list, "/var/run/daos_server"),
+        ("Client", client_list, "/var/run/daos_agent"),
+    )
+    for host_type, host_list, directory in file_checks:
+        status, nodeset = check_file_exists(host_list, directory)
+        if not status:
+            raise AgentFailed(
+                "{}: {} missing directory {} for user {}.".format(
+                    nodeset, host_type, directory, user))
```

Comment:
Carry over from the original code - and no, it does not.  The check should be more robust and check the file permissions using the user.  It may not gain much in CI but could be useful in manual testing for those not bitten once.

Reply:
Added verifying the user ownership as well as the path existence.